job/job.go

```diff
@@ -248,9 +248,41 @@ func (j *Job) Disable() {
 	j.Disabled = true
 }
 
+// DeleteFromParentJobs goes through and deletes the current job from any parent jobs.
+func (j *Job) DeleteFromParentJobs(cache JobCache) error {
+	j.lock.Lock()
+	defer j.lock.Unlock()
+	if len(j.ParentJobs) != 0 {
+		for _, p := range j.ParentJobs {
+			parentJob, err := cache.Get(p)
+			parentJob.lock.Lock()
+			defer parentJob.lock.Unlock()
```

Comment:
@ajvb Note that this won't unlock until the end of the function, not the end of the for-loop, which may or may not be OK.